battleground_unit_control/src/register_interface.rs

```diff
@@ -324,19 +324,12 @@ impl RegisterInterface {
 use crate::interface::InterfaceError;
 impl crate::interface::Interface for RegisterInterface {
     fn modules(&self) -> Result<Vec<u32>, BoxedError> {
-        Ok(self
-            .modules
-            .iter()
-            .map(|(module_index, _module)| *module_index)
-            .collect::<_>())
```

Comment:
Oh interesting, these clippies don't show up because the register interface is a conditional compilation and `cargo clippy --release` at the top level won't compile it.